CRM/Eck/BAO/EckEntityType.php

```diff
@@ -14,8 +14,23 @@
 +--------------------------------------------------------*/
 
 use CRM_Eck_ExtensionUtil as E;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Civi\Core\Event\PreEvent;
+use Civi\Core\Event\PostEvent;
+use Civi\Api4\CustomGroup;
+use Civi\Api4\EckEntity;
+use Civi\Api4\OptionValue;
+use Civi\Api4\Managed;
 
-class CRM_Eck_BAO_EckEntityType extends CRM_Eck_DAO_EckEntityType {
+class CRM_Eck_BAO_EckEntityType extends CRM_Eck_DAO_EckEntityType implements EventSubscriberInterface {
+
+  public static function getSubscribedEvents() {
+    return [
+      'hook_civicrm_pre::EckEntityType' => 'onPreEckEntityType',
+      'hook_civicrm_post::EckEntityType' => 'onPostEckEntityType',
```

Comment:
Because this BAO class (CRM_Eck_BAO_EckEntityType) matches the entity you want to target (EckEntityType), when switching to the classic naming convention with `HookInterface` you can use the magic name `self_hook_civicrm_pre/post` [as in this example](https://github.com/civicrm/civicrm-core/blob/d40fda2f63870384801e3267f12e92cd51c7ba9b/CRM/Campaign/BAO/Survey.php#L232).